pkg/plugin.go

```diff
@@ -79,25 +79,32 @@ func newDatasource() datasource.ServeOpts {
 }
 
 // QueryData handles multiple queries and returns multiple responses.
-// req contains the queries []DataQuery (where each query contains RefID as a unique identifer).
+// req contains the queries []DataQuery (where each query contains RefID as a unique identifier).
 // The QueryDataResponse contains a map of RefID to the response for each query, and each response
 // contains Frames ([]*Frame).
 func (td *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	// create response struct
 	response := backend.NewQueryDataResponse()
+	results := make(chan func(backend.Responses), len(req.Queries))
 
-	// loop over queries and execute them individually.
+	// Execute the queries in parallel and collect the responses.
 	for _, q := range req.Queries {
-		res := td.query(ctx, q)
-		// save the response in a hashmap
-		// based on with RefID as identifier
-		response.Responses[q.RefID] = res
+		go func(dq backend.DataQuery) {
+			res := doQuery(ctx, dq)
```

Comment:
do we need to handle errors?

Reply:
The response object has an `Error` field, but you got me to 👁️ it better and there is a place that should be covered better 👍